File: src/main/java/su/nlq/idea/externalizable/modification/AddImports.java
package su.nlq.idea.externalizable.modification;

import com.intellij.openapi.project.Project;
import com.intellij.psi.CommonClassNames;
import com.intellij.psi.PsiClass;
import com.intellij.psi.PsiFile;
import com.intellij.psi.PsiJavaFile;
import com.intellij.psi.codeStyle.CodeStyleSettings;
import com.intellij.psi.codeStyle.CodeStyleSettingsManager;
import com.intellij.psi.impl.source.codeStyle.ImportHelper;
import org.jetbrains.annotations.NotNull;

import java.io.IOException;
import java.io.ObjectInput;
import java.io.ObjectOutput;
import java.util.Arrays;
import java.util.Optional;

public final class AddImports implements PsiClassModification {
  @NotNull
  private final PsiJavaFile file;

  public AddImports(@NotNull PsiFile file) {
    this.file = (PsiJavaFile) file;
  }

  @Override
  public void accept(@NotNull PsiClass psiClass) {
    final Project project = psiClass.getProject();
    final CodeStyleSettings settings = CodeStyleSettingsManager.getSettings(project);
    final ImportHelper importHelper = new ImportHelper(settings);

    Arrays.asList(
        CommonClassNames.JAVA_IO_EXTERNALIZABLE,
        ObjectInput.class.getName(),
        ObjectOutput.class.getName(),
        IOException.class.getName(),
        ClassNotFoundException.class.getName()
    ).stream()
        .map(className -> new TypeFinder(project, className))
        .map(TypeFinder::get)
        .map(type -> Optional.ofNullable(type.resolve()))
        .forEach(resolved -> resolved.ifPresent(reference -> importHelper.addImport(file, reference)));
  }
}

File: src/main/java/su/nlq/idea/externalizable/modification/AddImports.kt
package su.nlq.idea.externalizable.modification

import com.intellij.psi.CommonClassNames
import com.intellij.psi.PsiClass
import com.intellij.psi.PsiFile
import com.intellij.psi.PsiJavaFile
import com.intellij.psi.codeStyle.CodeStyleSettingsManager
import com.intellij.psi.impl.source.codeStyle.ImportHelper
import java.io.IOException
import java.io.ObjectInput
import java.io.ObjectOutput
import java.util.*

class AddImports(file: PsiFile) : PsiClassModification {
  private val file: PsiJavaFile

  init {
    this.file = file as PsiJavaFile
  }

  override fun accept(psiClass: PsiClass) {
    val project = psiClass.project
    val settings = CodeStyleSettingsManager.getSettings(project)
    val importHelper = ImportHelper(settings)

    Arrays.asList(
        CommonClassNames.JAVA_IO_EXTERNALIZABLE,
        ObjectInput::class.java.name,
        ObjectOutput::class.java.name,
        IOException::class.java.name,
        ClassNotFoundException::class.java.name
    ).stream()
        .map { TypeFinder(project, it) }
        .map { it.get() }
        .map { Optional.ofNullable(it.resolve()) }
        .forEach { it.ifPresent { reference -> importHelper.addImport(file, reference) } }
  }
}
